src/GojsDiagram.test.tsx

```diff
@@ -0,0 +1,150 @@
+    };
+
+    const myDiagramId = 'myDiagramId';
+
+    let diagram: Diagram;
+    let wrapper;
+
+    beforeEach(() => {
+        const dom = document.body;
+        wrapper = mount(
+            (
+                <GojsDiagram
+                    diagramId={myDiagramId}
+                    model={model}
+                    createDiagram={(id) => {
+                        diagram = createDiagram(myDiagramId);
+                        return diagram;
+                    }
+                    }
+                    className="fakecss"
+                />
+            ),
+            { attachTo: dom });
+    });
+
+    it('renders the model provided as parameter', () => {
+        checkRenderedModel(model);
+        wrapper.unmount();
+    });
+
+    const checkRenderedModel = (currentModel) => {
+        expect(diagram.nodes.count).toBe(currentModel.nodeDataArray.length);
+        currentModel.nodeDataArray.forEach(node => {
+            expect(diagram.nodes.any(e => e.key === node.key && e.data.color === node.color)).toBeTruthy();
+        });
+        currentModel.linkDataArray.forEach(link => {
+            expect(diagram.links.any(e => e.fromNode.key === link.from && e.toNode.key === link.to)).toBeTruthy();
+        });
+    };
+
+    const testInputs = [
+        {
+            updatedModel: {
+                ...model,
+                nodeDataArray: [
+                    ...model.nodeDataArray,
+                    { key: 'New', color: 'blue' }
+                ]
+            }
+        },
+        {
+            updatedModel: {
+                nodeDataArray: [
+                    { key: 'Alpha', color: 'lightblue' },
+                    { key: 'Beta', color: 'orange' },
+                    { key: 'Gamma', color: 'lightgreen' },
+                ],
+                linkDataArray:
+                    [
+                        { from: 'Alpha', to: 'Beta' },
+                        { from: 'Alpha', to: 'Gamma' }
+                    ]
+            }
+        },
+        {
+            updatedModel: {
+                ...model,
+                linkDataArray:
+                    [
+                        ...model.linkDataArray,
+                        { from: 'Alpha', to: 'Omega' }
+                    ]
+            }
+        },
+        {
+            updatedModel: {
+                ...model,
+                nodeDataArray: [
+                    { key: 'Alpha', color: 'lightblue' },
+                    { key: 'Beta', color: 'blue' },
+                    { key: 'Gamma', color: 'black' },
+                    { key: 'Delta', color: 'pink' },
+                    { key: 'Omega', color: 'red' }
+                ]
+            }
+        }
+    ];
+
+    testInputs.forEach(test => {
+        it('updates the render given a new model', () => {
+            checkRenderedModel(model);
+            wrapper.setProps({ model: test.updatedModel });
+            checkRenderedModel(test.updatedModel);
+            wrapper.unmount();
```

Comment:
it smells as it should be a different test case. you want to test if unmounting clears the rendered model, right?

Reply:
yes ! I've just created a new test dedicated to the unmount step